src/resource.ts

```diff
@@ -0,0 +1,345 @@
+import axios from "axios";
+import moment from "moment";
+import BlueButton from "./index";
+
+// generic error message
+const GENERAL_DATA_ERR =
+  '{"message": "Unable to load data - query FHIR resource error."}';
+
+const retrySettings = {
+  initInterval: 5,
+  maxAttempts: 3,
+  backOffExpr: "interval * ( 2 ** i )",
+  endpointPattern: "^/v[12]/fhir/.*",
+  retryableCodes: [500, 502, 503, 504],
+};
+
+export enum FhirResourceType {
+  Patient = "Patient",
+  Coverage = "Coverage",
+  Profile = "Profile",
+  ExplanationOfBenefit = "ExplanationOfBenefit",
+}
+
+export class AuthorizationToken {
+  public accessToken: string;
+  public expiresIn: number;
+  public expiresAt: number;
+  public tokenType: string;
+  public scope: [string];
+  public refreshToken: string;
+  public patient: string;
+
+  constructor(authToken: any) {
+    this.accessToken = authToken.access_token;
+    this.expiresIn = authToken.expires_in;
+    this.expiresAt = authToken.expires_at
+      ? authToken.expires_at
+      : moment().add(this.expiresIn).valueOf();
+    this.patient = authToken.patient;
+    this.refreshToken = authToken.refresh_token;
+    this.scope = authToken.scope;
+    this.tokenType = authToken.token_type;
+  }
+
+  isExpired(): boolean {
+    return moment(this.expiresAt).isBefore(moment());
+  }
+
+  validate(): void {
+    if (
+      !(
+        this.accessToken &&
+        this.expiresIn &&
+        this.expiresAt &&
+        this.patient &&
+        this.refreshToken &&
+        this.scope &&
+        this.tokenType
+      )
+    ) {
+      throw new Error("Invalid authorization token.");
+    }
+  }
+}
+
+export class FhirRequest {
```

Comment:
I don't think `FhirRequest` needs to be a class as written. It's not really providing any functionality - it's simply a data holder. It would probably be better as just a type. Or to move functionality into this class instead. Like it might have a `request` function that accepts `config` and then does the actual request.

Reply:
remember introduced FhirRequest per previous review comment..., right, it is actually used to encapsulate related states and methods ..., will see how to re-factor to best serve the sdk client code...

Reply:
now FhirRequest removed